aptos-move/framework/aptos-stdlib/sources/data_structures/big_ordered_map.move

```diff
@@ -443,6 +449,36 @@ module aptos_std::big_ordered_map {
         new_iter(prev_index, child_iter, iter_key)
     }
 
+    /// Apply the function to each element in the vector, consuming it.
+    public(friend) inline fun for_each<K: drop + copy + store, V: store>(self: BigOrderedMap<K, V>, f: |K, V|) {
+        // TODO - this can be done more efficiently, by destroying the leaves directly
+        // but that requires more complicated code and testing.
+        let it = self.new_begin_iter();
+        while (!it.iter_is_end(&self)) {
+            let k = *it.iter_borrow_key();
+            let v = self.remove(&k);
+            f(k, v);
+            it = self.new_begin_iter();
+        };
+        destroy_empty(self)
+    }
+
+    /// Apply the function to a reference of each element in the vector.
+    public(friend) inline fun for_each_ref<K: drop + copy + store, V: store>(self: &BigOrderedMap<K, V>, f: |&K, &V|) {
+        let it = self.new_begin_iter();
+        while (!it.iter_is_end(self)) {
+            f(it.iter_borrow_key(), it.iter_borrow(self));
+            it = it.iter_next(self);
+        };
+    }
+
+    /// Destroy a map, by destroying elements individually.
+    public(friend) inline fun destroy<K: drop + copy + store, V: store>(self: BigOrderedMap<K, V>, dv: |V|) {
```

Comment:
public(friend) cannot be invoked from aptos-framework. So need to make them public?

Reply:
Also since it's an inline function you probably don't need the friend?

Reply:
Can you remove public inline function freely?